ERCS/erc-7595.md

```diff
@@ -0,0 +1,491 @@
+
+This EIP addresses various challenges encountered by the crypto community with [EIP-721](./eip-721.md) based collections and assets. This EIP brings forth advantages such as sustainable NFT royalties supported by tangible assets, an on-chain verifiable floor price, and the introduction of additional monetization avenues for NFT collection creators.
+
+### Presets
+
+* The Basic Preset allows for the evaluation of an on-chain verifiable price floor for a specified NFT asset.
+
+* The Dynamic Preset facilitates on-chain modification of tokenURI based on predefined collateral rules for a specified NFT asset.
+
+* With the Royalty Preset, NFT collection creators can receive royalty payments for each transaction involving asset owners and Externally Owned Accounts (EOA), as well as transactions with smart contracts.
+
+* The VRF Preset enables the distribution of collateral among multiple NFT asset holders using the Verifiable Random Function (VRF) by Chainlink.
+
+### Extension to Existing EIP-721 Based Collections
+
+For numerous [EIP-721](./eip-721.md) based collections that cannot be redeployed, we propose the implementation of an abstraction layer embodied by a smart contract. This smart contract would replicate all the functionalities of this EIP standard and grant access to collateral through mapping.
+
+## Specification
+### EIP standard for new NFT collections
+
+```solidity
+
+interface IERC721Envious is IERC721 {
+	event Collateralized(uint256 indexed tokenId, uint256 amount, address tokenAddress);
+	event Uncollateralized(uint256 indexed tokenId, uint256 amount, address tokenAddress);
+	event Dispersed(address indexed tokenAddress, uint256 amount);
+	event Harvested(address indexed tokenAddress, uint256 amount, uint256 scaledAmount);
+
+	/**
+	 * @dev An array with two elements. Each of them represents percentage from collateral
+	 * to be taken as a commission. First element represents collateralization commission.
+	 * Second element represents uncollateralization commission. There should be 3 
+	 * decimal buffer for each of them, e.g. 1000 = 1%.
+	 *
+	 * @param uint 256 index of value in array.
+	 */
+	function commissions(uint256 index) external view returns (uint256);
+
+	/**
+	 * @dev 'Black hole' is any address that guarantees that tokens sent to it will not be 
+	 * retrieved from it. Note: some tokens revert on transfer to zero address.
+	 *
+	 * @return address address of black hole.
+	 */
+	function blackHole() external view returns (address);
```

Comment:
How will ERC721 OnReceive behave for this address in this case? Is that behavior defined?

[onERC721Received](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-)

Reply:
The proposed standard is focused on ERC-20 based collateral only. The proposed standard doesn't include ERC-721 as collateral to an ERC-721 collection.